dxr/query.py

```diff
@@ -209,6 +231,43 @@ def direct_result(self):
                 elif len(results) > 1:
                     return None
 
+    def _search_and_highlight(self, query, filters, is_line_query, offset, limit,
+                              should_alphabetize=True):
+        """Perform a query using self.es_search and highlight the results,
+        returning a tuple (total count of results, list of results)
+
+        :arg query: valid ES query
+        :arg filters: nested list of filters
+        :arg is_line_query: whether to query lines (otherwise query files)
+        :arg offset: return results starting from this number
+        :arg limit: maximum number of results to return
+        :arg should_alphabetize: whether to should_alphabetize the results alphabetically by path and
+        numerically by line
+
+        """
+        # Decide on the sort mode: either alphanumerically or by score.
+        if should_alphabetize:
+            sort_mode = ['path', 'number'] if is_line_query else ['path']
+        else:
+            sort_mode = ['_score']
```

Comment:
I think we need a secondary sort field to avoid arbitrary ordering like this: 
![screen shot 2016-07-26 at 16 45 30](https://cloud.githubusercontent.com/assets/295816/17154874/d2e410a2-5350-11e6-9711-346286986e7d.png)